pytorch_lightning/utilities/data.py

```diff
@@ -40,8 +40,9 @@ def has_len(dataloader: DataLoader) -> bool:
     if has_len and has_iterable_dataset(dataloader):
         rank_zero_warn(
             'Your `IterableDataset` has `__len__` defined.'
-            ' In combination with multi-processing data loading (e.g. batch size > 1),'
-            ' this can lead to unintended side effects since the samples will be duplicated.'
+            ' In combination with multi-process data loading (when num_workers > 1),'
+            ' `__len__` could be inaccurate if each worker is not configured independently'
+            ' to avoid having duplicate data.'
```

Comment:
I don't quite get the message. It checks that `__len__` is not defined but then says `__len__` could be inaccurate...

How can it not be defined and inaccurate at the same time?

Reply:
It displays the warning if len is defined, not the other way around.

Reply:
oops im blind 🙈